File: content/Input Validation/CS2 C++/exampleInCode.cpp
#include <iostream>
using namespace std;

int main()
{
  unsigned int vals[10];
  size_t which;
  unsigned int square;

  for (size_t i = 0; i < 10; i++ )
    vals[i] = (i+1)*(i+1);

  cout << "Please type a number: ";
  cin >> which;

  square = vals[which-1];
  cout << "The square of " << which << " is " << square << endl;
  return 0;
}

File: content/Secure Coding/Input Validation/CS2 Java/exampleInCode.java
import java.util.Scanner;
public class InputValidationExample {

  public static void main(String[] args) {
    int[] vals = new int[10];

    for (int i = 0; i < 10; i++) {
      vals[i] = (i+1)*(i+1);
    }

    System.out.print("Please type a number: ");
    Scanner sc = new Scanner(System.in);
    int which = sc.nextInt();

    int square = vals[which-1];
    System.out.println("The square of "+which+" is "+square);
  }
}